pkg/apis/options/header.go

```diff
@@ -21,10 +21,12 @@ type Header struct {
 // make up the header value
 type HeaderValue struct {
 	// Allow users to load the value from a secret source
-	*SecretSource `json:",omitempty"`
+	// TODO(tuunit): Investigate why mapstructure: ",squash" doesn't work with pointers
+	SecretSource `json:",squash"`
 
 	// Allow users to load the value from a session claim
-	*ClaimSource `json:",omitempty"`
+	// TODO(tuunit): Investigate why mapstructure: ",squash" doesn't work with pointers
+	ClaimSource `json:",squash"`
```

Comment:
Keep the more verbose approach. Remove squashing and go back to the sources being pointers.